content/thursday/exercises_Bayesian_regularization.qmd

```diff
@@ -0,0 +1,449 @@
+
+```{r}
+fit <- lm(Rings ~ Length, data = train)
+summary(fit)
+```
+
+We can view this estimate as the general, unregularized estimate so using flat priors. You can check this by running the same analysis with `brms` and its default priors.
+
+```{r}
+#| message: false
+#| eval: false
+#| code-fold: true
+#| code-summary: "Show the code"
+fit <- brm(Rings ~ Length, data = train)
+summary(fit)
+```
+
+Now consider a normal shrinkage prior with a standard deviation of $\sigma = 10$. This prior looks like this:
+
+```{r}
+plot(density(rnorm(100000, 0, sd = 10)))
+```
+
+You can see that most of the prior mass lies between -20 and 20. How much influence do you think this prior will exert on the regression coefficient of `Length`, that is estimated to be around 14 when using flat priors? You can check your intuition by running the analysis.
+
+```{r}
+#| message: false
+#| eval: false
+#| code-fold: true
+#| code-summary: "Show the code"
+fit <- brm(Rings ~ Length, data = train,
+           prior = set_prior("normal(0, 10)", class = "b"))
+summary(fit)
+# There is practically no influence of this prior; it is relatively uninformative given the scale of the variable.
+```
+
+Suppose that `Length` was measured in centimeters instead of millimeters. Let's create a new variable, `Length_cm`:
+
+```{r}
+train_new <- train
+train_new$Length_cm <- train_new$Length/10
+```
+
+Now, our normal(0, 10) prior is much more informative!
+
+```{r}
+#| message: false
+#| eval: false
+#| code-fold: true
+#| code-summary: "Show the code"
+fit1 <- lm(Rings ~ Length_cm, data = train_new)
+summary(fit1)
+
+fit2 <- brm(Rings ~ Length_cm, data = train_new,
+           prior = set_prior("normal(0, 10)", class = "b"))
+summary(fit2)
+# Now, the effect is shrunken heavily!
+```
+
+You can imagine that if we have variables measured on different scales, say one variable in cm and another in mm, that specifying one common shrinkage prior (or even a prior in general) is not a good idea because the informativeness of the prior, and thus the amount of shrinkage, will depend on the scale of the variable. Therefore, it is highly recommended to always standardize your variables before applying regularization. This makes it easier to specify one general shrinkage prior for all coefficients.
+
+Rerun the analysis with the original ridge shrinkage prior but this time first scaling the data.
+
+```{r}
+#| message: false
+#| code-fold: true
+#| code-summary: "Show the code"
+
+train_scaled <- data.frame(scale(train[, -1], center = TRUE, scale = TRUE))
+train_scaled$Sex <- train$Sex
+
+fit_ridge_scaled <- brm(Rings ~ ., data = train_scaled,
+             prior = set_prior(lasso(df = 1, scale = 1), class = "b"),
+             refresh = 0, silent = 2)
+summary(fit_ridge_scaled)
+# Now, we would select Shell_weight and Shucked_weight based on the 95% CI!
```

Comment:
These comments on what is seen are in the code. Might be nicer to have interpretations not in the code-blocks. But that might be a personal preference.